src/proto_io.rs

```diff
@@ -1,10 +1,11 @@
 //! Protobuf message IO task structure.
 
-use protobuf::{self, Message};
-use std::io::{Read, Write};
+use std::cmp;
+use std::io::{self, Read, Write};
 use std::marker::PhantomData;
 use std::net::TcpStream;
-use std::{cmp, io};
+
+use protobuf::{self, CodedOutputStream, Message, ProtobufError};
```

Comment:
Modules may have their own `Message` types. Referring to those types as `module::Message` adds readability, in my opinion.

Reply:
Ok, so you are saying that it would be more clear to write:

```
use protobuf::{self, CodedOutputStream, ProtobufError};
...
impl<S: Read + Write, M: protobuf::Message> ProtoIo<S, M>
...
```

Instead of:

```
use protobuf::{self, CodedOutputStream, Message, ProtobufError};
...
impl<S: Read + Write, M: Message> ProtoIo<S, M>
```

I agree with the argument that there are multiple `Message` types in the `hbbft` crate, and adding `protobuf::` before `Message` would make things clearer.